src/scheduler/job_info.c

```diff
@@ -3057,26 +3058,37 @@ find_jobs_to_preempt(status *policy, resource_resv *hjob, server_info *sinfo, in
 	if (hjob == NULL || sinfo == NULL)
 		return NULL;
 
+	/* use locally dup'd copy of sinfo so we don't modify the original */
+	if ((nsinfo = dup_server_info(sinfo)) == NULL) {
+		free_schd_error_list(full_err);
+		free(pjobs);
+		free_string_array(preempt_targets_list);
+		return NULL;
+	}
```

Comment:
By moving this call where you did, you'll now dup the server for every job we don't run.  The server was dup'd where it was for a reason.  Duping the server is expensive.  We want to dup it as close to where it is used as possible.  That will give the rest of the function every possibility to return before we dup.